### ISSUE TITLE
Inconsistent behavior of consecutiveBlankLines rule

### ISSUE THREAD

**Issue by rodion-artyukhin** [nicklockwood/SwiftFormat#1460]
Hi!

I'm writing to report an issue related to the `consecutiveBlankLines` rule. While using this rule, I've observed inconsistent behavior where it fails to throw a lint warning when there are two blank lines at the end of a file.

The expected behavior of the `consecutiveBlankLines` rule is to detect and warn about consecutive blank lines within a file. However, it appears to overlook the presence of two blank lines specifically at the end of a file.

To illustrate the issue further, let me provide an example:
```
final class NetworkService {
    ...
    <-- 1st blank line
    <-- 2nd blank line: consecutiveBlankLines rule works correctly and warns about the blank line
    ...
}
<-- 1st blank line
<-- 2nd blank line: consecutiveBlankLines rule doesn't throw a warning about the blank line
<-- 3rd blank line: consecutiveBlankLines rule starts acting correctly, throwing a warning
```

1. If a file contains two or more consecutive blank lines within the scope of a type or a function, the `consecutiveBlankLines` rule correctly throws a lint warning indicating the presence of excessive blank lines.
2. However, if the file concludes with exactly two consecutive blank lines, the `consecutiveBlankLines` rule fails to raise any lint warning. This contradicts the expected behavior, as it should ideally flag such occurrences consistently.

Here is the screenshot that illustrates the `Swiftformat` output after running the lint process:

![Swiftformat](https://github.com/nicklockwood/SwiftFormat/assets/132341635/860110e0-3e63-49d2-a965-8c2880948681)

Thanks in advance for investigating this issue.

**Comment by nicklockwood** [comment]
@rodion-artyukhin thanks for reporting this, it will be fixed in the next release.

The issue was that SwiftFormat was comparing the line before and after and seeing that nothing had changed (the blank line was still a blank line). It does this to avoid false positives in the lint output (where one rule might have made a change that is cancelled out by another) but obviously in this case it was doing the wrong thing.

**Comment by nicklockwood** [comment]
@rodion-artyukhin fixed in 0.51.11

**Comment by rodion-artyukhin** [comment]
@nicklockwood Thanks a lot for such a prompt fix!
